www/utils/supabase/admin.ts

```diff
@@ -21,6 +22,71 @@ const supabaseAdmin = createClient<Database>(
   process.env.SUPABASE_SERVICE_ROLE_KEY || ''
 );
 
+// Trial membership
+const createOrRetrieveFreeTrialSubscription = async (userId: string) => {
+  // Check for existing trial subscription
+  const { data: existingSub, error: subError } = await supabaseAdmin
+    .from('subscriptions')
+    .select('*')
+    .eq('user_id', userId)
+    .eq('status', 'trialing')
+    .maybeSingle();
+
+  if (subError) throw new Error(`Subscription lookup failed: ${subError.message}`);
+
+  // If trial subscription exists, return it
+  if (existingSub) return existingSub;
+
+  // Create a new trial subscription
+  const subscriptionData: TablesInsert<'subscriptions'> = {
+    id: `free_trial_${userId}`,
+    user_id: userId,
+    status: 'trialing',
```

Comment:
By manually inserting a subscription to the database we have a weird situation where there is no data in stripe linking the customer and Bloom user. Also our subscription check will get one subscription and if a user pays for a subscription, but also has a trialing subscription it'll lead to weird behavior.